Example/src/screens/HeaderOptions.tsx

```diff
@@ -0,0 +1,173 @@
+import React, {useState, useLayoutEffect, useEffect} from 'react';
+import {StyleSheet, ScrollView, Text} from 'react-native';
+import {
+  createNativeStackNavigator,
+  NativeStackNavigationProp,
+} from 'react-native-screens/native-stack';
+
+import {
+  SettingsInput,
+  SettingsPicker,
+  SettingsSwitch,
+  Square,
+  Button,
+} from '../shared';
+
+type StackParamList = {
+  Main: undefined;
+  Settings: undefined;
+};
+
+interface MainScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Main'>;
+}
+
+const MainScreen = ({navigation}: MainScreenProps): JSX.Element => {
+  useEffect(() => {
+    navigation.navigate('Settings');
+  }, []);
+
+  return (
+    <ScrollView>
+      <Button
+        onPress={() => navigation.navigate('Settings')}
+        title="Go to next screen"
+      />
+      <Button onPress={() => navigation.pop()} title="🔙 Back to Examples" />
+    </ScrollView>
+  );
+};
+
+type HeaderItemPosition = 'right' | 'center' | 'left';
+
+interface SettingsScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Settings'>;
+}
+
+const SettingsScreen = ({navigation}: SettingsScreenProps): JSX.Element => {
+  const [headerTitle, setHeaderTitle] = useState('Settings');
+  const [backButtonVisible, setBackButtonVisible] = useState(true);
+  const [headerShown, setHeaderShown] = useState(true);
+  const [headerLargeTitle, setHeaderLargeTitle] = useState(false);
+  const [headerItem, setHeaderItem] = useState<HeaderItemPosition>('right');
+  const [headerBackTitle, setHeaderBackTitle] = useState('Back');
+  const [headerHideShadow, setHeaderHideShadow] = useState(false);
+  const [headerTranslucent, setHeaderTranslucent] = useState(false);
+
+  const square = (props: {tintColor?: string}) => (
+    <Square {...props} color="green" size={20} />
+  );
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle,
+      headerHideBackButton: !backButtonVisible, // android
+      headerBackTitleVisible: backButtonVisible, // iOS
+      headerLargeTitle, // iOS
+      headerBackTitle, // iOS
+      headerShown,
+      headerRight: headerItem === 'right' ? square : undefined,
+      headerCenter: headerItem === 'center' ? square : undefined,
+      headerLeft: headerItem === 'left' ? square : undefined,
+      headerHideShadow,
+      headerTranslucent,
+    });
+  }, [
+    navigation,
+    headerTitle,
+    backButtonVisible,
+    headerLargeTitle,
+    headerBackTitle,
+    headerItem,
+    headerShown,
+    headerHideShadow,
+    headerTranslucent,
+  ]);
+
+  return (
+    <ScrollView style={styles.container}>
```

Comment:
```suggestion
    <ScrollView contentInsetAdjustmentBehavior="automatic" style={styles.container}>
```

Add this when you have a ScrollView with `native-stack` header.

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/827/commits/0ebaefd6685cec73fb9ddcd4ad5f7b76549de316